plugins/woocommerce/templates/order/order-details.php

```diff
@@ -83,7 +85,7 @@
 				?>
 					<tr>
 						<th scope="row"><?php echo esc_html( $total['label'] ); ?></th>
-						<td><?php echo ( 'payment_method' === $key ) ? esc_html( $total['value'] ) : wp_kses_post( $total['value'] ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?></td>
+						<td><?php echo ( 'payment_method' === $key ) ? wc_get_container()->get( HtmlSanitizer::class )->sanitize( $total['value'] ) : wp_kses_post( $total['value'] ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?></td>
```

Comment:
seems like we should be using `wp_kses_post` here instead of the sanitizer class to be consistent with  the [thankyou page](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/templates/checkout/thankyou.php#L71). HtmlSanitizer is already being run when getting the titime in `abstract-wc-payment-gateway.php`

Reply:
(Fixed in 3a1fc7744219562faf7ff033e80c0cd8eb38877f)